subprojects/packagefiles/openssl/meson.build

```diff
@@ -0,0 +1,457 @@
+    arch_subdir = 'linux64-mips64'
+  else
+    error('Unsupported arch+OS combo: ' + host_machine.cpu_family() + ' + ' + host_machine.system())
+  endif
+
+  message('OpenSSL is configured without ASM support')
+else
+  message('OpenSSL is configured with ASM support')
+endif
+
+subdir('generated-config/archs' / arch_subdir / asm)
+
+# Build options specific to OS
+if host_machine.system() == 'windows'
+  defines += [
+    ## default of Win. See INSTALL in openssl repo.
+    'OPENSSLDIR="C:\\Program Files\\Common Files\\SSL"',
+    'ENGINESDIR="NUL"',
+    'OPENSSL_SYS_WIN32', 'WIN32_LEAN_AND_MEAN', 'L_ENDIAN',
+    '_CRT_SECURE_NO_DEPRECATE', 'UNICODE', '_UNICODE',
+  ]
+  if compiler.get_id() == 'msvc'
+    c_args += [
+      '-wd4090', '-Gs0', '-GF', '-Gy', '-nologo',
+    ]
+    foreach library: openssl_libraries
+      dependencies += [
+        compiler.find_library(library),
+      ]
+    endforeach
+  else
+    foreach library: openssl_libraries
+      dependencies += [
+        compiler.find_library(
+          library,
+          static: get_option('default_library') == 'static',
+        ),
+      ]
+    endforeach
+  endif
+elif host_machine.system() == 'darwin'
+  defines += [
+    'OPENSSLDIR="/System/Library/OpenSSL/"',
+    'ENGINESDIR="/dev/null"',
+  ]
+  foreach library: openssl_libraries
+    dependencies += [
+      compiler.find_library(
+        library,
+        static: get_option('default_library') == 'static',
+      ),
+    ]
+  endforeach
+  c_args += [
+    '-Wno-missing-field-initializers',
+  ]
+elif host_machine.system() == 'sunos'
+  defines += [
+   'OPENSSLDIR="/etc/ssl"',
+   'ENGINESDIR="/dev/null"',
+   '__EXTENSIONS__'
+ ]
+  foreach library: openssl_libraries
+    dependencies += [
+      compiler.find_library(
+        library,
+        static: get_option('default_library') == 'static',
+      ),
+    ]
+  endforeach
+else
+  # linux and others
+  defines += [
+    'OPENSSLDIR="/etc/ssl"',
+    'ENGINESDIR="/dev/null"',
+    'TERMIOS',
+  ]
+  foreach library: openssl_libraries
+    dependencies += [
+      compiler.find_library(
+        library,
+        static: get_option('default_library') == 'static',
+      ),
+    ]
+  endforeach
+  c_args += [
+    '-Wno-missing-field-initializers',
+  ]
+  if compiler.get_id() != 'clang'
+    c_args += [
+      '-Wno-old-style-declaration',
+    ]
+  endif
+endif
+
+foreach dependency: openssl_dependencies
+  dependencies += [dependency(dependency)]
+endforeach
+
+# We may need to add some defines for static builds
+if get_option('default_library') == 'static'
+  defines += [
+    'OSSL_CRYPTO_DSO_CONF_H',
+    'DSO_NONE',
+    'DSO_EXTENSION=.so',
+    'OPENSSL_NO_DSO',
+  ]
+endif
+
+foreach define : defines + openssl_defines
+  c_args += ['-D' + define]
+endforeach
+
+c_args += openssl_cflags
+
+include_directories += openssl_include_directories
+
+libcrypto_lib = library(
+  'libcrypto',
+  name_prefix: '',
```

Comment:
The usual convention is to name a library e.g. "crypto" and use a name_prefix that is per-platform, e.g. on linux it is 'lib' and on Windows it is ''.

I genuinely have no idea, but... why does it need the 'lib' everywhere?

(Also I'm not sure but I somehow feel like it might be more correct to name it 'crypto' and use name_prefix: 'lib', I wonder if there's a general style preference one way or another used by other projects facing the same challenge.)

Reply:
This is what packages are called on Linux, this is what those libraries are referred to on https://wiki.openssl.org/index.php/Main_Page

I'm not sure what the correct way to name those is, but it felt natural to use the same names for compilation targets.

Reply:
The correct way is to do `library('crypto', ...)` and not put in any of your own prefixes of suffixes. Meson will name the output files correctly for all platforms automatically.

Reply:
Right, if you don't have any specific reason to think it *must* be still called "libcrypto" even on Windows, then you should definitely go with the platform defaults.